app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -502,6 +506,8 @@ protected void onPause() {
         if (adapter != null) {
             adapter.unregisterDataSetObserver(mListStateObserver);
         }
+
+        hereFunctionHandler.stopLocationUpdates();

```

Comment:
Yeah, it depends on whether `loader` is `null` or not and if there are any pending tasks in `EntityLoaderTask`. I don't fully understand what is going on because that is a very circular piece of logic. I think that part of the code is trying to ensure entity loading continues if the entity select activity is destroyed temporarily (on screen rotation)